docs: Add Helm docs for tls.existingSecretName (#11306)
gravitational/teleport: docs/pages/kubernetes-access/helm/reference.mdx

  </TabItem>
</Tabs>

## `tls.existingSecretName`

| Type | Default value | Can be used in `custom` mode? | `teleport.yaml` equivalent |
| - | - | - | - |
| `string` | `""` | ✅ | `proxy_service.https_keypairs` |

`tls.existingSecretName` tells Teleport to use an existing Kubernetes TLS secret to secure its web UI using HTTPS. This can be

> set to use a TLS certificate issued by a trusted internal CA rather than a public-facing CA like LetsEncrypt.

Reviewer: ```suggestion
set to use a TLS certificate issued by a trusted internal CA rather than a public-facing CA like Let's Encrypt.
```

> You should create the secret in the same namespace as Teleport using `kubectl create secret tls my-tls-secret --cert=/path/to/cert/file --key=/path/to/key/file`

Reviewer: I think we should turn this command into a `code` snippet so it's copyable

See https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets for more information.

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  tls:
    existingSecretName: my-tls-secret
  ```
  </TabItem>
  <TabItem label="--set">
  ```shell
  --set tls.existingSecretName=my-tls-secret
  ```
  </TabItem>
</Tabs>

## `tls.existingSecretNameCA`

| Type | Default value | Can be used in `custom` mode? |
| - | - | - |
| `string` | `""` | ✅ |

> `tls.existingSecretNameCA` sets the `SSL_CERT_FILE` environment variable to load a trusted CA or bundle in PEM format into Teleport pods.

Reviewer: Is it just the Auth Service that reads this environment variable?

Also, will readers be expected to understand the use of `SSL_CERT_FILE` more than (or before they) understand `tls.existingSecretNameCA`? If not, I think we can make this paragraph clearer by removing the mention of the env var.

Author: > Is it just the Auth Service that reads this environment variable?

It's the proxy service that's potentially reading it here to establish a full trust chain when serving Teleport's TLS listener.

I mostly put this here for competent readers who might Ctrl+F SSL_CERT_FILE against the reference to see whether we have some way to override this. It's fairly well documented online that `SSL_CERT_FILE` and `SSL_CERT_DIR` are some ways to solve this - I figured it wouldn't hurt if people are deep enough into PKI.

> This can be set to inject a root and/or intermediate CA so that Teleport can build a full trust chain on startup. This is likely to be needed

Reviewer: ```suggestion
This can be set to inject a root and/or intermediate CA so that Teleport can build a full trust chain on startup. 

This is likely to be needed
```

This paragraph is on the long side—suggested splitting it.

if Teleport fails to start when `tls.existingSecretName` is set with a `User Message: unable to verify HTTPS certificate chain` error
in the pod logs.

> You should create the secret in the same namespace as Teleport using `kubectl create secret generic my-root-ca --from-file=ca.pem=/path/to/root-ca.pem`

Reviewer: Same as above re: the command

> <Admonition type="note" title="Root CA filename">

Reviewer: Since this Admonition has a single line, I think it might work better as a `Notice` with type `warning`.

>   The filename used for the root CA in the secret is important - it must be `ca.pem`.

Reviewer: ```suggestion
  The filename used for the root CA in the secret must be `ca.pem`.
```
